### ISSUE TITLE
Pyenv not using currently active python's pip command

### ISSUE THREAD

**Issue by avillamarin-plenty** [pyenv/pyenv#2348]
### Description
- [x] Platform information: Ubuntu 22.04
- [x] OS architecture: amd64 
- [x] pyenv version: 2.2.5
- [x] Python version: 3.9.12 (installed with pyenv)
- [x] C Compiler information: 11.2.0
- [x] Please attach the debug trace of the failing command as a gist: There is no specific faulty command
  
I have the following problem...sort to speak..

By the way did some research and the closes I could found before creating this one is this https://github.com/pyenv/pyenv/issues/2206

I've installed pyenv in a fresh Ubuntu 22.04 couple days back and only installed one python 3.9 using it.

```
pyenv versions
* system (set by /home/internetmosquito/.pyenv/version)
3.9.12

```

Now I noticed when switching to system python (3.10) it seems

```
python3
Python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>

```

That the pip used is one referenced in my home folder at .local

```
which pip
/home/internetmosquito/.local/bin/pip
which pip3
/home/internetmosquito/.local/bin/pip3
```

I honestly don't even recall when or how these 2 got there...maybe while installing pipenv? In any case, I wonder why pyenv won't switch to the pip which are living in /usr/lib/python3/dist-packages...by the way I actually seem to have 2 directories there, one for python3 and another for python3.10...

```
ls -ltra /usr/lib/python3/dist-packages | grep pip
drwxr-xr-x   2 root root   4096 abr 25 16:01 pip-22.0.2.dist-info
drwxr-xr-x   5 root root   4096 abr 25 16:01 pip
```

So, now if I enable again the python 3.9 with pyenv

```
pyenv global 3.9.12
python
Python 3.9.12 (main, Apr 25 2022, 16:30:54) 
[GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>
which pip
/home/internetmosquito/.local/bin/pip
```

By the way, I have another pip in where python 3.9 lives

```
ls -ltra /home/internetmosquito/.pyenv/versions/3.9.12/bin/ | grep pip         
-rwxrwxr-x 1 internetmosquito internetmosquito      264 abr 25 16:31 pip3.9
-rwxrwxr-x 1 internetmosquito internetmosquito      264 abr 25 16:31 pip3
lrwxrwxrwx 1 internetmosquito internetmosquito        6 abr 25 16:31 pip -> pip3.9
```

So basically ...

1.- Why pyenv does not enable the pip that is living inside the python 3.9 installed?

**Comment by native-api** [comment]
Please check your PATH and inspect which exact executables run in each case which `which` and `pyenv which`.

Looks like you shell configuration isn't quite right and Pyenv's shims are sometimes being shadowed by other executables.

---

For the record, we have just released 2.3.0 that simplifies shell setup and improved the README.

**Comment by avillamarin-plenty** [comment]
This is my path

```
$ echo $PATH
/home/internetmosquito/tmp/pcli:/home/internetmosquito/.poetry/bin:/home/internetmosquito/.global-node-modules/bin:/home/internetmosquito/.pyenv/plugins/pyenv-virtualenv/shims:/home/internetmosquito/.pyenv/bin:/home/internetmosquito/.local/bin:/home/internetmosquito/.sdkman/candidates/java/current/bin:/home/internetmosquito/.nvm/versions/node/v12.22.12/bin:/home/internetmosquito/.pyenv/shims:/home/internetmosquito/.pyenv/bin:/home/internetmosquito/.poetry/bin:/home/internetmosquito/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/internetmosquito/tmp/chromedriver:/home/internetmosquito/tmp/geckodriver

```

Output of which pyenv

```
pyenv () {
	local command
	command="${1:-}" 
	if [ "$#" -gt 0 ]
	then
		shift
	fi
	case "$command" in
		(activate | deactivate | rehash | shell) eval "$(pyenv "sh-$command" "$@")" ;;
		(*) command pyenv "$command" "$@" ;;
	esac
}

```

What I have related with pyenv in my .zshrc file:

```
export PATH="/home/internetmosquito/.pyenv/bin:$PATH"
eval "$(pyenv init -)"
eval "$(pyenv virtualenv-init -)"
export PYENV_ROOT=/home/internetmosquito/.pyenv

```

**Comment by avillamarin-plenty** [comment]
@native-api that helped, thanks a lot! Please close this as you see fit

**Comment by vedal** [comment]
I observed that `pyenv-virtualenv` linked to `/usr/bin/pip` although I had specified `pyenv global 3.8.5`. What solved the problem was an update of virtualenv:

`pip install -U virtualenv`
